src/core/jpeg_stream.js

```diff
@@ -53,8 +43,24 @@ class JpegStream extends DecodeStream {
   }
 
   readBlock() {
+    return this.decodeImage();
+  }
+
+  decodeImage(bytes) {
     if (this.eof) {
-      return;
+      return this.buffer;
+    }
+    if (!bytes) {
+      bytes = this.bytes;
+    }
+
+    // Some images may contain 'junk' before the SOI (start-of-image) marker.
+    // Note: this seems to mainly affect inline images.
+    for (let i = 0, ii = bytes.length - 1; i < ii; i++) {
+      if (bytes[i] === 0xff && bytes[i + 1] === 0xd8) {
+        bytes = bytes.subarray(i);
```

Comment:
Do we want to skip this operation when it's unnecessary?
```suggestion
        if (i > 0) {
          bytes = bytes.subarray(i);
        }
```